src/web/mod.rs

```diff
@@ -134,7 +134,7 @@ fn match_version(
                  WHERE normalize_crate_name(name) = normalize_crate_name($1)",
                 &[&name],
             )
-            .unwrap(); // FIXME: remove this unwrap when all handlers using it are migrated to axum
+            .context("error fetching crate")?;
```

Comment:
Is it worth adding the data about what crate is being searched for? (or is that added to the error at a higher level?)

```suggestion
            .with_context(|| format!("error fetching normalized crate {name:?}")?;
```

Reply:
tracing add its spans, so we definitely see all the request details (see [here](https://rust-lang.sentry.io/issues/3972510218/) for an example)

Reply:
I would expect a panic to get the same spans attached too. I'm guessing the issue is that the `match_version` call is `spawn_blocking`'d off and loses the tracing context. `utils::spawn_blocking` should do something like

```rust
let span = tracing::Span::current();
tokio::task::spawn_blocking(|| span.in_scope(f))
    .await
    .context("failed to join thread")?
```

Reply:
valid point. 

I tried the approach above, but it still didn't show the request context for panics. 

I think this needs more digging, but I believe this shoudln't block this PR.

Reply:
Please add a FIXME and/or open an issue so it's not forgotten then.

Reply:
I was intrigued and started digging into this. 

First of all: 
tokio already does `catch_unwind` and will return a `JoinError` in case of a panic. 

So we will have _both_ the panic and a normal server error (with the needed URL context): 

<img width="555" alt="grafik" src="https://user-images.githubusercontent.com/540890/223533942-c16eb1e8-eb35-4c39-aacf-cf93f853f0d5.png">

Reply:
( funny: the `panic!` message will not be in the join-error :) )

Reply:
so, it's a combination, but the `panic` message actually comes from the global `sentry-panic` handler, which doesn't have / use the tracing integration

Reply:
So, I believe I found a possible solution, but I'm happy to discuss. 

- `spawn_blocking` will resume the panic when it got one 
- a newly added `CatchPanicLayer` will convert panics into server errors (until now, these lead to closed connections). 

This wouldn't get rid of the duplication between the global panic-handler, and the error in the web handlers. But the errors from the web handler would at least contain all the needed context.